src/core/lib/security/authorization/evaluate_args.h

```diff
@@ -33,9 +29,22 @@ namespace grpc_core {
 
 class EvaluateArgs {
  public:
-  EvaluateArgs(grpc_metadata_batch* metadata, grpc_auth_context* auth_context,
-               grpc_endpoint* endpoint)
-      : metadata_(metadata), auth_context_(auth_context), endpoint_(endpoint) {}
+  // Caller is responsible for ensuring auth_context outlives PerChannelArgs
+  // struct.
+  struct PerChannelArgs {
+    PerChannelArgs(grpc_auth_context* auth_context, grpc_endpoint* endpoint);
+
+    absl::string_view transport_security_type;
+    absl::string_view spiffe_id;
```

Comment:
DNS name will be plumbed in a different PR?

Reply:
DNS name plumbing PR is already submitted. Since I submitted that today, after this PR was approved, I thought I will add it in a follow-up PR.